Routers/Routers/ConfigurationGenerator.cs

```diff
@@ -0,0 +1,145 @@
+﻿// <copyright file="ConfigurationGenerator.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace Routers;
+
+/// <summary>
+/// Network Configuration Generator.
+/// </summary>
+public class ConfigurationGenerator
+{
+    private readonly Dictionary<int, List<(int, int)>> topology;
+    private readonly Dictionary<int, List<(int, int)>> resultTopology;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ConfigurationGenerator"/> class.
+    /// </summary>
+    /// <param name="path">A path to the input data file.</param>
+    public ConfigurationGenerator(string path)
+    {
+        this.topology = new Dictionary<int, List<(int, int)>>();
+        this.resultTopology = new Dictionary<int, List<(int, int)>>();
+        this.ParseTopology(path);
+        this.GenerateMaxSpanningTree();
+    }
+
+    /// <summary>
+    /// Writes the built configuration to the output file.
+    /// </summary>
+    /// <param name="outputPath">The output file path.</param>
+    public void WriteOutput(string outputPath)
+    {
+        using var writer = new StreamWriter(outputPath);
+        foreach (int router in this.resultTopology.Keys.Where(r => this.resultTopology[r].Any()).OrderBy(r => r))
+        {
+            var connections = this.resultTopology[router]
+                .OrderBy(c => c.Item1)
+                .Select(c => $"{c.Item1} ({c.Item2})");
+            var enumerable = connections.ToList();
+
+            writer.WriteLine($"{router}: {string.Join(", ", enumerable)}");
+        }
+    }
+
+    private void ParseTopology(string inputPath)
+    {
+        string[] data = File.ReadAllLines(inputPath);
+        if (data.Length == 0)
+        {
+            throw new NullGraphException();
+        }
+
+        foreach (string line in data)
+        {
+            string trimmedString = line.Trim();
+            if (string.IsNullOrEmpty(trimmedString))
+            {
+                continue;
+            }
+
+            string[] parts = trimmedString.Split(':');
+            if (parts.Length != 2)
+            {
+                throw new FormatException();
```

Comment:
Желательно здесь сообщать, в чём проблема:
```suggestion
                throw new FormatException("Line must contain 2 parts, got {parts.Length}");
```

Вообще, системные исключения без сообщения используются очень редко.